code/modules/mob/say.dm

```diff
@@ -111,14 +111,16 @@
 */
 
 /mob/proc/say_quote(var/message, var/datum/language/speaking = null)
-        var/verb = "says"
-        var/ending = copytext(message, length(message))
-        if(ending=="!")
-                verb=pick("exclaims","shouts","yells")
-        else if(ending=="?")
-                verb="asks"
-
-        return verb
+	. = "says"
+	var/ending = copytext(message, length(message))
+	var/pre_ending = copytext(message, length(message) - 1, length(message))
+	if(ending == "!")
+		if(pre_ending == "!" || pre_ending == "?")
+			. = pick("shouts", "yells")
```

Comment:
Shouldn't these verbs be based off the language verbs? And same on 121. Why are they hard-coded here?

Reply:
Infact, shouldn't this proc just call /datum/language/proc/get_spoken_verb(var/msg_end, var/pre_end)?